src/psij/testing/service.py

```diff
@@ -384,29 +384,31 @@ def run(self, site_id, run_id) -> object:
 
     @cherrypy.expose
     @cherrypy.tools.json_out()
-    def getLatestResultForTestOnAllSites(self, sitesToGet) -> object:
+    def tests(self, sites_to_get, tests_to_match) -> object:
 
         import json
-        sitesToGet = json.loads(sitesToGet)
+        sites_to_get = json.loads(sites_to_get)
+        tests_to_match = json.loads( tests_to_match)
 
-        # Example sitesToGet = ["mothra.hidden.uoregon.edu", "reptar.hidden.uoregon.edu", "saturn.hidden.uoregon.edu"]
+        # Example sites_to_get = ["mothra.hidden.uoregon.edu", "reptar.hidden.uoregon.edu", "saturn.hidden.uoregon.edu"]
         resp = {}
 
-        for site_id in sitesToGet:
-            resp[site_id] = self.getSite(site_id)
+        for site_id in sites_to_get:
+            resp[site_id] = self.getSite(site_id, tests_to_match)
 
         return resp
 
 
-    def getSite(self, site_id):
+    def getSite(self, site_id, tests_to_match):
         #  Example site_id="mothra.hidden.uoregon.edu"
         run_id = ""
 
-        runs = RunEnv.objects(site_id=site_id).order_by('+test_start_time')
+        runs = RunEnv.objects(site_id=site_id).order_by('-test_start_time')[0:1]
 
         for run in runs:
             run_id = run.run_id
```

Comment:
You could replace those three lines with

```Python
run_id = RunEnv.objects(site_id=site_id).order_by('-test_start_time')[0].run_id
```

Reply:
I tried this out, unfortunately it doesn't work.  I get "IndexError: no such item for Cursor instance" I think it needs to iterate through the for loop to get the first run.

Reply:
That's most likely because you are asking it to give you runs for a site that doesn't exist, so the returned list will have zero elements. So you might want to catch that IndexError and return an empty response in that case. 

The code you have now will attempt to iterate through the list, but, since it's empty for a non-existent site, it will not and run_id will be left at the default value of "".

Reply:
Ok, that worked.  Thanks!